devicetypes/smartthings/philio-multiple-sound-siren.src/philio-multiple-sound-siren.groovy

```diff
@@ -0,0 +1,314 @@
+	"Forever": 0,
+	"30 seconds": 1,
+	"1 minute": 2,
+	"2 minutes": 4,
+	"3 minutes": 6,
+	"5 minutes": 10,
+	"10 minutes": 20,
+	"20 minutes": 40,
+	"30 minutes": 60,
+	"45 minutes": 90,
+	"1 hour": 120
+]}
+
+def getDefaultSound() { "Emergency" }
+def getDefaultDuration() { "Forever" }
+
+def installed() {
+	// Device-Watch simply pings if no device events received for 32min(checkInterval)
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+
+	state.sound = defaultSound
+	state.duration = defaultDuration
+
+	// Get default values
+	response([secure(zwave.basicV1.basicGet()), secure(zwave.configurationV1.configurationSet(parameterNumber: 31, size: 1, configurationValue: [durationMap[state.duration]]))])
+}
+
+def updated() {
+	def commands = []
+	// Device-Watch simply pings if no device events received for 32min(checkInterval)
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+
+	log.debug "settings: ${settings.inspect()}, state: ${state.inspect()}"
+
+	def sound = settings.sound ?: state.sound
+	def duration = settings.duration ?: state.duration
+
+	if (sound != state.sound || duration != state.duration) {
+		state.sound = sound
+		state.duration = duration
+		commands << secure(zwave.configurationV1.configurationSet(parameterNumber: 31, size: 1, configurationValue: [durationMap[duration]]))
+	}
+
+	response(commands)
+}
+
+def parse(String description) {
+	log.debug "parse($description)"
+	def result = null
+	if (description.startsWith("Err")) {
+		if (zwInfo?.zw?.contains("s")) {
+			result = createEvent(descriptionText:description, displayed:false)
+		} else {
+			result = createEvent(
+					descriptionText: "This device failed to complete the network security key exchange. If you are unable to control it via SmartThings, you must remove it from your network and add it again.",
+					eventType: "ALERT",
+					name: "secureInclusion",
+					value: "failed",
+					displayed: true,
+			)
+		}
+	} else {
+		def cmd = zwave.parse(description, [0x98: 1, 0x20: 1, 0x70: 1])
+		if (cmd) {
+			result = zwaveEvent(cmd)
+		}
+	}
+	log.debug "Parse returned ${result?.inspect()}"
+	return result
+}
+
+def zwaveEvent(securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand([0x20: 1, 0x85: 2, 0x70: 1])
+	// log.debug "encapsulated: $encapsulatedCommand"
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	}
+}
+
+def zwaveEvent(basicv1.BasicReport cmd) {
+	log.debug "rx $cmd"
+	handleDeviceEvent(cmd.value)
+}
+
+def zwaveEvent(sensorbinaryv2.SensorBinaryReport cmd) {
+	log.debug "rx $cmd"
+	def result = []
+
+	if (cmd.sensorType == sensorbinaryv2.SensorBinaryReport.SENSOR_TYPE_TAMPER) {
+		result << createEvent(name: "tamper", value: "detected")
+	} else {
```

Comment:
Is this the case for Siren State Reports? If so, Could we check that the sensorType to be general purpose? and discard other types?

Reply:
We only receive these events for Sensor Binary Report:

SensorBinaryReport(sensorValue: 8, sensorType: 1)
SensorBinaryReport(sensorValue: 255, sensorType: 1)
SensorBinaryReport(sensorValue: 0, sensorType: 1)

If it is a tamper event, we get:

SensorBinaryReport(sensorValue: 255, sensorType: 8)
SensorBinaryReport(sensorValue: 255, sensorType: 1)
And then if the alarm duration is anything other than forever, we get at the end:
SensorBinaryReport(sensorValue: 0, sensorType: 1)
SensorBinaryReport(sensorValue: 0, sensorType: 1)

Otherwise it is:
SensorBinaryReport(sensorValue: 255, sensorType: 1)
duration timeout if not forever, then:
SensorBinaryReport(sensorValue: 0, sensorType: 1)

Reply:
`SensorBinaryReport(sensorValue: 8, sensorType: 1)` Which one is that one? Out of curiosity?

Reply:
Bleh. I made a typo on that. It should be SensorBinaryReport(sensorValue: 255, sensorType: 8)